### DISCUSSION TITLE
AC current coil

### DISCUSSION THREAD

**Discussion by dizcza** [magpylib/magpylib#561] [Q&A] [answered]
I'm building a real Helmholtz coil that accepts DC - 1000 Hz AC input. Reactive impedances come into play.
Is it possible to model AC coils in magpy?

**Comment by OrtnerMichael** [comment]
Hi Danylo,

This depends on what your aim is. Magpylib computes the magnetic field, given a current. if your current has a time-dependence, e.g. i(t) = i0 * sin(wt), then the magnetic field will be B(t) = B0 * sin(wt). The field B0 is the static field generated by the current i0.

If you are interested in computing the coil impedances, Magpylib can help, but you have to make the computation by yourself. The impedance is a result of the coil self-interaction. This comes down to computing the flux through the coil, generated by an input current. For that Magpylib can help when combined with a numerical integrator, but it requires a lot of handwork.

Does this answer your question ?

sincerely,
Michael

**Reply by dizcza** [reply]
> if your current has a time-dependence, e.g. i(t) = i0 * sin(wt), then the magnetic field will be B(t) = B0 * sin(wt)

I'm not sure about it because impedance comes into play and at high frequencies capacitance as well. So I'm not sure if there are phase shifts in the resulting vector **B** compared to the input current phase. And the amplitude changes as well.

> For that Magpylib can help when combined with a numerical integrator, but it requires a lot of handwork.

That's what I was afraid of. I googled "frequency" in your library and found no mentions. I think magpylib is great for DC and AC modeling is beyond its scope.

**Comment by OrtnerMichael** [comment, marked as answer]
i(t) is the current in the system. If you inject a current, the above relation will be correct - its simply the relation between current in the system and magnetic field generated. If you drive with a voltage, there will be a frequency dependent phase shift an amplitude modulation because of the frequency dependent impedance.

To use Magpylib in the AC regime you have to do some hands-on modelling.

sincerely,
michael
